x-pack/plugins/observability/public/hooks/use_fetch_last24h_alerts.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState, useCallback } from 'react';
+import { AsApiContract } from '@kbn/actions-plugin/common';
+import { HttpSetup } from '@kbn/core/public';
+import { BASE_RAC_ALERTS_API_PATH } from '@kbn/rule-registry-plugin/common/constants';
+import { RULE_LOAD_ERROR } from '../pages/rule_details/translations';
+
+interface UseFetchLast24hAlertsProps {
+  http: HttpSetup;
+  features: string;
+  ruleId: string;
+}
+interface FetchLast24hAlerts {
+  isLoadingLast24hAlerts: boolean;
+  last24hAlerts: number;
+  errorLast24hAlerts: string | undefined;
+}
+
+export function useFetchLast24hAlerts({ http, features, ruleId }: UseFetchLast24hAlertsProps) {
```

Comment:
I will take example on this hook https://github.com/elastic/kibana/blob/main/x-pack/plugins/cases/public/containers/use_get_cases_status.tsx, so you can also take advantage of the abortController and the didCancel techniques to avoid some react error if user changes their mind by going to another page before this request finished.